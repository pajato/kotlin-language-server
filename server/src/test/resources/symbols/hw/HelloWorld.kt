package symbols.hw

class HelloWorld() {
    fun main() {
        println("Hello Kotlin Language Server")
    }
}
