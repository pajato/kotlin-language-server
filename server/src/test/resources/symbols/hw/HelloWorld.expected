DocumentSymbol("||path-to-symbols||/hw/HelloWorld.kt", File, KtFile, package symbols.hw@@class HelloWorld() {@    fun main() {@        println("Hello Kotlin Language Server")@    }@}@)
    DocumentSymbol("hw", Package, KtPackageDirective, package symbols.hw, symbols.hw)
        DocumentSymbol("symbols.hw", Package, KtDotQualifiedExpression, symbols.hw)
    DocumentSymbol("HelloWorld", Class, KtClass, class HelloWorld() {@    fun main() {@        println("Hello Kotlin Language Server")@    }@})
        DocumentSymbol("HelloWorld", Constructor, KtPrimaryConstructor, ())
        DocumentSymbol("", Class, KtClassBody, {@    fun main() {@        println("Hello Kotlin Language Server")@    }@})
            DocumentSymbol("main", Function, KtNamedFunction, fun main() {@        println("Hello Kotlin Language Server")@    })
                DocumentSymbol("()", Function, KtParameterList, ())
                DocumentSymbol("main", Function, KtBlockExpression, {@        println("Hello Kotlin Language Server")@    })
                    DocumentSymbol("main", Function, KtCallExpression, println("Hello Kotlin Language Server"))
                        DocumentSymbol("println", Function, KtNameReferenceExpression, println)
                        DocumentSymbol("main", Function, KtValueArgumentList, ("Hello Kotlin Language Server"))
